msm/src/serialization/mod.rs

```diff
@@ -102,11 +145,83 @@ mod tests {
                     constraints.push(cst.clone())
                 }
             }
+
+            for (j, lookup) in witness_env.rangecheck4_lookups.iter().enumerate() {
+                rangecheck4[j].push(lookup.clone())
+            }
+
+            for (j, lookup) in witness_env.rangecheck15_lookups.iter().enumerate() {
+                rangecheck15[j].push(lookup.clone())
+            }
+
+            witness_env.add_rangecheck4_table_value(i);
```

Comment:
I feel like this needs to be called somewhere from `range_check4()` function on the witness `impl InterpreterEnv` side, no?

Reply:
I would like to move it somewhere else, yes. The table should be built outside of the interpreter as it is a fixed table. We should have a way to handle multiple domain size.